includes/Modules/Analytics_4.php

```diff
@@ -1402,8 +1402,16 @@ private function get_custom_dimensions_data() {
 						$data[ $custom_dimension ] = $post->post_author;
 						break;
 					case 'googlesitekit_post_categories':
-						$cats                      = wp_get_post_categories( $post->ID, array( 'fields' => 'ids' ) );
-						$data[ $custom_dimension ] = ! is_wp_error( $cats ) ? implode( ',', $cats ) : '';
+						$categories = get_the_category( $post->ID );
+
+						if ( ! empty( $categories ) ) {
+							$get_category_ids = function( $category ) {
+								return $category->term_id;
+							};
+
+							$data[ $custom_dimension ] = implode( ',', array_map( $get_category_ids, $categories ) );
```

Comment:
We can simplify this to use core wp function - `wp_list_pluck` instead of `array_map`.
```suggestion
$category_ids = wp_list_pluck( $categories, 'term_id' );

$data[ $custom_dimension ] = implode( ',', $category_ids );
```

Reply:
Ah, right on, good one. I tend to forget about the WP functions 😆